From 4c86609815d6e4f053c5fe51640c726004befc4d Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Thu, 27 May 2021 16:33:18 +0300
Subject: [PATCH] [SW-43588] dw: spi: mask interrupts

When initializing the SPI HW, we need to mask the interrupts in IMR
register. Seems like the interrupts are unmasked by mistake, as 0 means
masked and the IMR is set with 1s.

Change-Id: I5ea5ea8bd3f1b61837a4900f286c0320e87102fe
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/common/spi.c b/common/spi.c
index 0228506..0711fae 100644
--- a/common/spi.c
+++ b/common/spi.c
@@ -273,7 +273,7 @@
 	regmap_writel(baudr, spi_regs + BAUDR);
 	regmap_writel(0x00000000, spi_regs + RXFTLR);
 	regmap_writel(0x00000000, spi_regs + TXFTLR);
-	regmap_writel(0x0000003F, spi_regs + IMR);	/* Disable Interrupts */
+	regmap_writel(0x00000000, spi_regs + IMR);	/* Disable Interrupts */
 	regmap_writel(0x00000001, spi_regs + SER);
 	regmap_writel(0x00000001, spi_regs + SSIENR);
 
